test-common/utils.c

```diff
@@ -2169,11 +2169,14 @@ eos_test_setup_flatpak_repo_with_preinstalled_apps (GFile        *updater_dir,
    * so the remote collection ID needs to be set after the flatpak is
    * installed. FIXME: The tests should be using GPG in order to be
    * representative of a real system. */
-  if (!set_flatpak_remote_collection_id (updater_dir,
-                                         repo_name,
-                                         collection_id,
-                                         error))
-    return FALSE;
+  if (collection_id != NULL)
```

Comment:
The `collection_id` is used in the `flatpak_populate_runtime()` call above, which claims to be setting up the repository as well as the remote config. Is it OK for it to be `NULL` there? Would it make more sense to have two function arguments: one for the collection ID as used in the repository, and one for it as configured in the remote config?

Reply:
The `collection_id` gets passed from there to `ostree_init` - that can handle a `NULL` `collection_id` just fine, it just won't pass it to ostree when it sets up the repo:

```
gboolean
ostree_init (GFile *repo,
             RepoMode mode,
             const gchar *collection_id,
             CmdResult *cmd,
             GError **error)
{
  CmdArg args[] =
    {
      { NULL, "init" },
      { "mode", repo_mode_to_string (mode) },
      { (collection_id != NULL) ? "collection-id" : NULL, collection_id },
      { NULL, NULL }
    };

  return spawn_ostree_in_repo_args (repo,
                                    args,
                                    cmd,
                                    error);
```

I suppose it could be useful to set up a `collection_id` on the repo and not on the remote.

Reply:
> I suppose it could be useful to set up a `collection_id` on the repo and not on the remote.

I’d say it’s necessary to test both, since we have both setups. Most of our repositories on endlessm.com have collection IDs; some don’t.

Reply:
Okay. I've split up the configuration, though I have yet to add a test for the latter case.